src/module/actor/character/feats/types.ts

```diff
@@ -19,21 +20,57 @@ interface FeatLike<TParent extends ActorPF2e | null = ActorPF2e | null> extends
 interface FeatGroupData {
     id: string;
     label: string;
-    featFilter?: string[];
     supported?: FeatOrFeatureCategory[];
-    slots?: (FeatSlotCreationData | string | number)[];
+    /** Compendium browser filter properties when searching. Only category is validated */
+    filter?: FeatBrowserFilterProps;
+    slots?: (FeatSlotData | number)[];
+    /** If true, all slots are sorted by their level. Only applies if slotless */
+    sorted?: boolean;
+    /** If set to true, all slots except the first will be hidden unless its been assigned to */
+    requiresInitial?: boolean;
+    /** Default placeholder text for empty slots */
+    placeholder?: string;
+    /** If given, this is feat group has a configurable limit independent of level */
+    customLimit?: {
+        label: string;
+        min: number;
+        max: number;
+    } | null;
 }
 
-interface FeatSlot<TItem extends FeatLike | HeritagePF2e = FeatPF2e> {
-    id: string;
+interface FeatBrowserFilterProps {
+    traits?: string[];
+    conjunction?: "or" | "and";
+}
+
+/** Data used to describe a feat slot when creating a new feat group */
+interface FeatSlotData {
+    /**
+     * A globally unique id. In future versions this may become locally unique instead.
+     * If omitted, it is <group id>-<level>.
+     */
+    id?: string;
+    /** Short label for the feat slot, displayeed on the left. By default it is the level */
     label?: Maybe<string>;
+    /** The level of the feat that should go in this slot */
+    level?: Maybe<number>;
+    /**
+     * The limit value the group needs in order to display this feat.
+     * This can be different from level if the group is based on tiers.
+     */
+    rank?: Maybe<number>;
```

Comment:
I'm shopping for a better variable name if anyone can think of anything.

Reply:
`notRank` would be slightly better than that

Reply:
Renamed to `tier`. I was looking for a catch all term for the group's level, but its only ever going to tier or level anyways (even if its not necessarily the actor's level) so tier is fine.